tests/run/cpp_scoped_enums.pyx

```diff
@@ -0,0 +1,127 @@
+# tag: cpp,cpp11
+# mode: run
+
+from libcpp.limits cimport numeric_limits
+
+cdef extern from *:
+    """
+    enum class Enum1 {
+        Item1,
+        Item2
+    };
+    """
+    cdef enum class Enum1:
+        Item1
+        Item2
+
+
+cdef extern from * namespace "Namespace1":
+    """
+    namespace Namespace1 {
+        enum class Enum2 {
+            Item1,
+            Item2
+        };
+    }
+    """
+    cdef enum class Enum2:
+        Item1
+        Item2
+
+
+cdef enum class Enum3(int):
+    a = 1
+    b = 2
+
+
+cdef extern from *:
+    """
+    enum class sorted
+    {
+        a = 1,
+        b = 0
+    };
+    """
+    cdef enum class Enum4 "sorted":
+        a
+        b
+
+
+cdef extern from *:
+    """
+    #include <limits>
+
+    enum class LongIntEnum : long int {
+        val = std::numeric_limits<long int>::max(),
+    };
+    """
+    enum class LongIntEnum(long int):
+        val
+
+
+def test_compare_enums():
+    """
+    >>> test_compare_enums()
+    (True, True, False, False)
+    """
+    cdef Enum1 x, y
+    x = Enum1.Item1
+    y = Enum1.Item2
+
+    return (
+        x == Enum1.Item1,
+        y == Enum1.Item2,
+        x == Enum1.Item2,
+        y == Enum1.Item1
+    )
+
+        
+def test_compare_namespace_enums():
+    """
+    >>> test_compare_enums()
+    (True, True, False, False)
+    """
+    cdef Enum2 z, w
+    
+    z = Enum2.Item1
+    w = Enum2.Item2
+
+    return (
+        z == Enum2.Item1,
+        w == Enum2.Item2,
+        z == Enum2.Item2,
+        w == Enum2.Item1
+    )
+
+
+def test_coerce_to_from_py_value():
```

Comment:
You should probably also check the failure modes here. What happens if you try to assign a Python `int(3)` to enum2 (I suspect probably nothing exciting)? What happens if you try to assign `int(111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111)` (hopefully a Python overflow error)?

Reply:
Added those tests.